components/match2/wikis/teamfortress/match_summary.lua

```diff
@@ -0,0 +1,145 @@
+---
+-- @Liquipedia
+-- wiki=teamfortress
+-- page=Module:MatchSummary
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local DateExt = require('Module:Date/Ext')
+local Icon = require('Module:Icon')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local Page = require('Module:Page')
+
+local DisplayHelper = Lua.import('Module:MatchGroup/Display/Helper')
+local MatchSummary = Lua.import('Module:MatchSummary/Base')
+
+local htmlCreate = mw.html.create
+
+---@enum TFMatchIcons
+local Icons = {
+	CHECK = Icon.makeIcon{iconName = 'winner', color = 'forest-green-text', size = 'initial'},
+	EMPTY = '[[File:NoCheck.png|link=]]',
+}
+
+local LINK_DATA = {
+	logstf = {icon = 'File:Logstf_icon.png', text = 'logs.tf Match Page '},
+	logstfgold = {icon = 'File:Logstf_gold_icon.png', text = 'logs.tf Match Page (Golden Cap) '},
+	esl = {
+		icon = 'File:ESL 2019 icon lightmode.png',
+		iconDark = 'File:ESL 2019 icon darkmode.png',
+		text = 'ESL matchpage'
+	},
+	esea = {icon = 'File:ESEA icon allmode.png', text = 'ESEA Match Page'},
+	etf2l = {icon = 'File:ETF2L.png', text = 'ETF2L Match Page'},
+	rgl = {icon = 'File:RGL_Logo.png', text = 'RGL Match Page'},
+	ozf = {icon = 'File:ozfortress-icon.png‎', text = 'ozfortress Match Page'},
+	tftv = {icon = 'File:Teamfortress.tv.png', text = 'TFTV Match Page'},
+}
+
+local CustomMatchSummary = {}
+
+---@param args table
+---@return Html
+function CustomMatchSummary.getByMatchId(args)
+	return MatchSummary.defaultGetByMatchId(CustomMatchSummary, args)
+end
+
+---@param match MatchGroupUtilMatch
+---@param footer MatchSummaryFooter
+---@return MatchSummaryFooter
+function CustomMatchSummary.addToFooter(match, footer)
+	footer = MatchSummary.addVodsToFooter(match, footer)
+		:addLinks(LINK_DATA, match.links)
+
+	return footer
+end
+
+---@param match MatchGroupUtilMatch
+---@return MatchSummaryBody
+function CustomMatchSummary.createBody(match)
+	local body = MatchSummary.Body()
+
+	if match.dateIsExact or DateExt.isDefaultTimestamp(match.timestamp) then
```

Comment:
```suggestion
	if match.dateIsExact or not DateExt.isDefaultTimestamp(match.timestamp) then
```
I assume? Looks very wrong otherwise
But at this time you might as well just use
```suggestion
	if not DateExt.isDefaultTimestamp(match.timestamp) then
```

or am I missing a scenario?

Reply:
I cant recall any edge case related to Date Input as well, so probably no weird scenarios being left forgotten here